tabby-core/src/services/profiles.service.ts

```diff
@@ -211,27 +349,192 @@ export class ProfilesService {
         return null
     }
 
-    getConfigProxyForProfile <T extends Profile> (profile: PartialProfile<T>, skipUserDefaults = false): T {
+    /*
+    * Methods used to interract with Profile/ProfileGroup/Global defaults
+    */
+
+    /**
+    * Return global defaults for a given profile provider
+    * Always return something, empty object if no defaults found
+    */
+    getProviderDefaults (provider: ProfileProvider<Profile>): any {
+        const defaults = this.config.store.profileDefaults
+        return defaults[provider.id] ?? {}
+    }
+
+    /**
+    * Set global defaults for a given profile provider
+    */
+    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+    setProviderDefaults (provider: ProfileProvider<Profile>, pdefaults: any): void {
+        this.config.store.profileDefaults[provider.id] = pdefaults
+    }
+
+    /**
+    * Return defaults for a given profile
+    * Always return something, empty object if no defaults found
+    * arg: skipUserDefaults -> do not merge global provider defaults in ConfigProxy
+    */
+    getProfileDefaults (profile: PartialProfile<Profile>, skipGlobalDefaults = false, skipGroupDefaults = false): any[] {
```

Comment:
since this is a public api, it needs to be future proof by using an optional `options` object arg instead of separate params

Reply:
Resolved in 5e5c80832de2d2125a9e55d89321071365c32938